pleskdistup/actions/extensions.py

```diff
@@ -32,3 +37,88 @@ def estimate_post_time(self) -> int:
+
+        if not os.path.isfile(apache_config) or not files.find_file_substrings(apache_config, "PassengerRuby"):
+            # Likely it means ruby application is not enabled, so we need to do direct search in the filesystem
+            return None
+
+        application_root = files.find_file_substrings(apache_config, "PassengerAppRoot")
+        if not application_root:
+            return None
+
+        # Record format is: PassengerAppRoot "[path]"
+        # We can expect it is always in this format since configuration is generated by plesk
+        return application_root[0].split()[1].strip("\"")
+
+    def _get_ruby_application_path_by_bundle_subdir(self, domain_home: str) -> str:
+        bundle = files.find_subdirectory_by(domain_home,
+                                            lambda subdir: os.path.basename(subdir) == "bundle" and
+                                                           os.path.exists(os.path.join(subdir, "ruby")))
+        if bundle is None or not os.path.isdir(bundle):
+            return None
+
+        return os.path.dirname(os.path.dirname(bundle))
+
+    def _get_ruby_application_paths(self, domain_path: str) -> typing.Tuple[str, str]:
+        app_directory = self._get_ruby_application_path_by_config(domain_path.name)
+        if app_directory is None:
+            log.debug("Ruby application is disabled. Search for application root in filesystem. Domain home directory: {}".format(domain_path))
+            app_directory = self._get_ruby_application_path_by_bundle_subdir(domain_path)
+
+        if app_directory is None:
+            return None, None
+
+        return app_directory, os.path.join(app_directory, "vendor", "bundle")
+
+    def _post_action(self) -> action.ActionResult:
+        ruby_domains = (domain_path for domain_path in os.scandir("/var/www/vhosts") if self._is_ruby_domain(domain_path))
+        for domain_path in ruby_domains:
+            log.debug("Re-bundling ruby application in domain: {}".format(domain_path.name))
+
+            app_directory, bundle = self._get_ruby_application_paths(domain_path)
+
+            if bundle is None or not os.path.isdir(bundle):
+                log.debug("Skip re-bundling for non bundling domain '{}'".format(domain_path.name))
+                continue
+
+            stat_info = os.stat(app_directory)
+            username = pwd.getpwuid(stat_info.st_uid).pw_name
+
+            log.debug("Bundle: {}. App directory: {}. Username: {}".format(bundle, app_directory, username))
+
+            shutil.rmtree(bundle)
+            util.logged_check_call(["/usr/sbin/plesk", "sbin", "rubymng", "run-bundler", username, app_directory])
+
+        return action.ActionResult()
+
+    def _revert_action(self) -> action.ActionResult:
+        return action.ActionResult()
+
+    def estimate_post_time(self) -> int:
+        return 60 * len([domain_path for domain_path in os.scandir("/var/www/vhosts") if self._is_ruby_domain(domain_path)])
```

Comment:
Listing Ruby domains probably should be extracted into a function, because it's repeated multiple times above.